src/watchman/api_v1_handler.clj

```diff
@@ -65,7 +66,13 @@
       (do
         (models/remove-host-from-role (:id host) (:id role))
         {:status 204}) ; No content
-      (create-json-error-response 404 "hostname not found in role"))))
+      (create-json-error-response 404 "hostname not found in role")))
+
+  (POST "/snooze" {:keys [params role]}
+    (let [snooze-duration (-> params :duration Long/parseLong)
+          role-id (:id role)]
+      (models/snooze-role role-id snooze-duration)
+      (redirect (str "/roles/" role-id)))))
```

Comment:
Ya, I was sorta split on this point.  The more I think about it, the more I agree with you that a redirect on an API endpoint is weird.  I'll change it to AJAX